### PR TITLE
Adds Bokeh 3.5 enhancements to RangeToolLink

### LINKED ISSUE #6340: Add recent Bokeh 3.5 enhancements to RangeToolLink
Bokeh 3.5 added some new features to the RangeTool, including:
- [box select handles](https://github.com/bokeh/bokeh/pull/13906) (with `range_tool.overlay.use_handles = True`)
-  being able to [double click to start range of a new selection box](https://github.com/bokeh/bokeh/pull/13855). (with `RangeTool(start_gesture='tap')` or `range_tool.start_gesture='tap'`)
- [Inversion of box annotation](https://github.com/bokeh/bokeh/pull/13810) (with `range_tool.overlay.inverted = True`)

This is a request to enable these features by default in HoloViews with Bokeh backend. 

<details> <summary> Bokeh example </summary>

```python
import numpy as np
from bokeh.layouts import column
from bokeh.models import ColumnDataSource, RangeTool
from bokeh.plotting import figure, show
from bokeh.io import output_notebook

output_notebook()

n_points = 3000
x_values = np.linspace(0, 100, n_points)
y_values = np.random.randn(n_points).cumsum()

source = ColumnDataSource(data=dict(x=x_values, y=y_values))

detailed_plot = figure(
    width=800,
    height=300,
    tools=["xpan", "xzoom_in", "xzoom_out", "reset", "wheel_zoom"],
    toolbar_location="above",
    active_scroll="wheel_zoom",
    # background_fill_color="#efefef",
    x_range=(22, 30),
)

detailed_plot.line("x", "y", source=source)

minimap = figure(
    width=detailed_plot.width,
    height=150,
    tools="",
    toolbar_location=None,
    background_fill_color=detailed_plot.background_fill_color,
    title="Drag the middle or edges of the selection box below, or double click to start a new box"
)

minimap.line("x", "y", source=source)
minimap.x_range.range_padding = 0
minimap.ygrid.grid_line_color = None

range_tool = RangeTool(x_range=detailed_plot.x_range, y_range=detailed_plot.y_range)
range_tool.start_gesture = 'tap'
range_tool.overlay.fill_color = "grey"
range_tool.overlay.fill_alpha = 0.2
range_tool.overlay.inverted = True
range_tool.overlay.use_handles = True
range_tool.overlay.handles.all.hover_fill_color = "grey"
range_tool.overlay.handles.all.hover_fill_alpha = 0.25
range_tool.overlay.handles.all.hover_line_alpha = 0
range_tool.overlay.handles.all.fill_alpha = 0
range_tool.overlay.handles.all.line_alpha = 0.07
minimap.add_tools(range_tool)

show(column(detailed_plot, minimap))
```

</details>

https://github.com/bokeh/bokeh/assets/6613202/34d228fb-b920-4cef-99aa-3b0cfeaaeca7

### PR DESCRIPTION
**Pull request by droumis**
Closes #6340

https://github.com/user-attachments/assets/e0a1b70f-1051-43aa-8f93-f36fb154cb2b

### CODE REVIEW

**Comment by droumis** [comment]
@holoviz-developers, I'm seeking feedback on default styling of box handles and inverted highlighting

**Comment by ahuang11** [comment]
Styling looks good to me!

File: holoviews/plotting/bokeh/links.py [hunk 1]
```diff
@@ -182,6 +182,24 @@ def __init__(self, root_model, link, source_plot, target_plot):
...
+        if use_handles and bokeh35:
+            tool.overlay.use_handles = use_handles
+            tool.overlay.handles.all.hover_fill_color = "grey"
+            tool.overlay.handles.all.hover_fill_alpha = 0.25
+            tool.overlay.handles.all.hover_line_alpha = 0
+            tool.overlay.handles.all.fill_alpha = 0.1
+            tool.overlay.handles.all.line_alpha = 0.25
+
+        start_gesture = getattr(link, 'start_gesture', 'tap')
+        if start_gesture is not None and bokeh35:
```
Comment (ahuang11) [inline holoviews/plotting/bokeh/links.py:196]: Would this ever be None if the getattr default is tap?

File: holoviews/plotting/bokeh/links.py [hunk 2]
```diff
@@ -182,6 +182,24 @@ def __init__(self, root_model, link, source_plot, target_plot):
...
+            tool.start_gesture = start_gesture
+
+        inverted = getattr(link, 'inverted', True)
+        if inverted is not None and bokeh35:
```
Comment (ahuang11) [inline holoviews/plotting/bokeh/links.py:200]: Same here

File: holoviews/plotting/bokeh/links.py (hunk 1 above)
Comment (droumis) [reply holoviews/plotting/bokeh/links.py]: thanks for the feedback. I think you're right, I'm not sure from which condition I was previously thinking that None might be the value. I committed a change

File: holoviews/plotting/bokeh/links.py (hunk 2 above)
Comment (droumis) [reply holoviews/plotting/bokeh/links.py]: same as above

File: holoviews/plotting/bokeh/links.py [hunk 3]
```diff
@@ -182,6 +182,23 @@ def __init__(self, root_model, link, source_plot, target_plot):
...
+        use_handles = getattr(link, 'use_handles', True)
+        start_gesture = getattr(link, 'start_gesture', 'tap')
+        inverted = getattr(link, 'inverted', True)
+
+        if bokeh35:
+            tool.overlay.use_handles = use_handles
+            tool.start_gesture = start_gesture
+            tool.overlay.inverted = inverted
+
+        if use_handles:
```
Comment (hoxbro) [inline holoviews/plotting/bokeh/links.py:195]:
```suggestion
        if bokeh_35 and use_handles:
```

File: holoviews/plotting/bokeh/links.py [hunk 4]
```diff
@@ -182,6 +182,23 @@ def __init__(self, root_model, link, source_plot, target_plot):
...
+        use_handles = getattr(link, 'use_handles', True)
+        start_gesture = getattr(link, 'start_gesture', 'tap')
+        inverted = getattr(link, 'inverted', True)
+
+        if bokeh35:
+            tool.overlay.use_handles = use_handles
+            tool.start_gesture = start_gesture
+            tool.overlay.inverted = inverted
+
+        if bokeh_35 and use_handles:
```
Comment (hoxbro) [inline holoviews/plotting/bokeh/links.py:195]:
```suggestion
        if bokeh35 and use_handles:
```

**Comment by droumis** [comment]
@hoxbro , is this good to merge?

**Review by hoxbro — COMMENTED** [review]
Does this change the default behavior?

If it does, I think we need to warn people if they have a version lower than Bokeh 3.5 and actively set these new parameters.

File: holoviews/plotting/links.py [hunk 5]
```diff
@@ -115,8 +115,16 @@ class RangeToolLink(Link):
...
 
-    _requires_target = True
+    use_handles = param.Boolean(default=True, doc="""
+        Whether to display handles""")
+
+    start_gesture = param.ObjectSelector(default='tap', objects=['pan', 'tap', 'none'],
+                                         doc="""Gesture to start a range selection""")
 
+    inverted = param.Boolean(default=True, doc="""
+                             Whether to invert the highlighting of the range selection""")
```
[the remark this answers is not in the document]
Comment (droumis) [reply holoviews/plotting/links.py]: I added a note in each of the docstrings

**Comment by droumis** [comment]
> If it does, I think we need to warn people if they have a version lower than Bokeh 3.5 and actively set these new parameters.

I added a warning if bokeh34. Is this what you suggested?

File: holoviews/plotting/bokeh/links.py [hunk 6]
```diff
@@ -182,6 +182,24 @@ def __init__(self, root_model, link, source_plot, target_plot):
                     ax.reset_end = end
 
         tool = RangeTool(**axes)
+
+        if bokeh35:
+            use_handles = getattr(link, 'use_handles', True)
+            start_gesture = getattr(link, 'start_gesture', 'tap')
+            inverted = getattr(link, 'inverted', True)
+
+
```
Comment (hoxbro) [inline holoviews/plotting/bokeh/links.py:191]:
```suggestion
```

**Pull request merged**